src/pipx/standalone_python.py

```diff
@@ -119,7 +119,7 @@ def _unpack(full_version, download_link, archive: Path, download_dir: Path):
             )
 
         with tarfile.open(archive, mode="r:gz") as tar:
-            tar.extractall(download_dir)
+            tar.extractall(download_dir, filter="data")
```

Comment:
I think `filter` is only available in 3.12+? We need a compatibility layer here.

Reply:
Indeed, but that doesn't seem to be the cause of the CI errors.